lab/src/raft/append_entries.go

```diff
@@ -18,20 +18,26 @@ type AppendEntriesReply struct {
 	Success bool
 }
 
+func (rf *Raft) sendAppendEntries(server int, args *AppendEntriesArgs, reply *AppendEntriesReply) bool {
+	ok := rf.peers[server].Call("Raft.AppendEntries", args, reply)
+	return ok
+}
+
 func (rf *Raft) AppendEntries(args *AppendEntriesArgs, reply *AppendEntriesReply) {
 	rf.Lock()
 	defer rf.Unlock()
 
 	reply.Term = rf.currentTerm
 
-	// TODO(ling): Implement heartbeat.
-
 	//  Reply false if term < currentTerm (§5.1)
 	if args.Term < rf.currentTerm {
 		reply.Success = false
 		return
 	}
 
+	rf.leader == args.Leader
+	rf.lastHeartBeat = time.Now()
+
```

Comment:
If the current server is candidate should convert to follower

Reply:
Done.